### PR TITLE
Fixes #12584: Add DSC content to Rudder by example

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/12584

### CODE REVIEW

File: rudder-by-example/src/SUMMARY.adoc [hunk 1]
```diff
@@ -3,5 +3,9 @@
 . link:files/README.adoc[Files management]
 .. link:files/advanced-file-templating.adoc[Advanced file templating]
 .. link:files/edition-one-line.adoc[Enforce a line is present in a file only once]
+. link:application/README.adoc[Application management]
+.. link:application/install-exe-or-msi.adoc[Ensure the presence of a windows software]
```
Comment (amousset) [inline rudder-by-example/src/SUMMARY.adoc]: Windows

File: rudder-by-example/src/SUMMARY.adoc [hunk 2]
```diff
@@ -3,5 +3,9 @@
...
+. link:system/README.adoc[System management]
+.. link:system/manage-registries.adoc[Manage Windows registries with Rudder]
```
Comment (amousset) [inline rudder-by-example/src/SUMMARY.adoc]: remove "with Rudder"

File: rudder-by-example/src/SUMMARY.adoc (hunk 1 above)
Comment (amousset) [reply rudder-by-example/src/SUMMARY.adoc]: "Software deployment on Windows"?

File: rudder-by-example/src/SUMMARY.adoc (hunk 2 above)
Comment (amousset) [reply rudder-by-example/src/SUMMARY.adoc]: "Manage Windows registry content"?

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 3]
```diff
@@ -0,0 +1,78 @@
+= Install a software from a .exe or .msi binary
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: A program ("software" is uncountable)

I would also replace "binary" by "installer".

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+== Usecase
+
+When you need to make sure a given software is installed and its version is
+recent enough, you should be using the "Windows Software" technique.
+
+== Parameters
+Let's pretend we need to install 7zip on the latest version on ours agent.
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: "our agents"?

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+We will use the "Windows Software" technique to do that.
+
+The technique offers the following parameters:
+
+image::windows_software_technique.png[Windows Software Technique]
+
+Where:
+
+* `Name of the software` will be the software name, as Windows understand it.
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: understands

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+* `Path` is the relative path of the installer (.exe or .msi file), on the server side.
+* `Options` are parameters that will be given to the installer wizard.
+* `Minimal version` is the minimal version of the software that we expect on the node.
+
+
+
+The technique beahviour is quite simple:
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: behaviour

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+- If the software is not found on the node or its version is lower to the expected one, Rudder will execute the installer wizard with the given parameters and the given .exe/.msi file.
+- If the version found is newer, Rudder will report that the software is already installed on a sufficient version.
+
+== Setup
+The main task here is to find corrects parameters to pass to Rudder. The best way to do so
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: correct

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+_For #msi installer: Use the /q parameters#._
+
+We will try that by executing in a Powershell console:
+
+----
+msiexec /i "<path-to-7zip-msi-file>" /q
+----
+
+7zip should now be installed! We can now read from the Windows Registries our missing parameters from the output of the following
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: Registry

File: rudder-by-example/src/system/manage-registries.adoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
+= Manage Windows Registry keys
+
+== Usecase
+
+Windows softwares often come with some configuration directly taken form the Windows registries.
```
Comment (amousset) [inline rudder-by-example/src/system/manage-registries.adoc]: software

Comment (amousset) [reply rudder-by-example/src/system/manage-registries.adoc]: comes

File: rudder-by-example/src/system/manage-registries.adoc [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+A clear and correct registry management can only improve the software and system management of
```
Comment (amousset) [inline rudder-by-example/src/system/manage-registries.adoc]: I don't get the meaning of this sentence ("can only"?)

Maybe something like "The software and system configuration of your Windows nodes hence relies on a clear and correct registry management"

File: rudder-by-example/src/system/manage-registries.adoc [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+your Windows based nodes.
+
+== Base
+Rudder offers different way to manage the Windows registries:
```
[the remark this answers is not in the document]
Comment (amousset) [reply rudder-by-example/src/system/manage-registries.adoc]: I think "Windows Registry" is always singular.

File: rudder-by-example/src/system/manage-registries.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+* By using the Technique Editor and dedicated generic methods.
+* By using the technique `Registry key edition`
+
+The technique is only a fancy overlay of the generic method.
```
Comment (amousset) [inline rudder-by-example/src/system/manage-registries.adoc]: methods

File: rudder-by-example/src/system/manage-registries.adoc [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+|Binary |Binary data
+
+|===
+
+
+NOTE: Non Char based type values must be entered via hexadecimal syntax: like 0xfff3f2f
+
+
+== Usage
+There are two types of generici methods to handle registries:
```
Comment (amousset) [inline rudder-by-example/src/system/manage-registries.adoc]: generic

Comment (amousset) [reply rudder-by-example/src/system/manage-registries.adoc]: registry keys

**Comment by amousset** [comment]
Besides some little details to fix, this is great :+1:

**Comment by Fdall** [comment]
Commit modified

File: rudder-by-example/src/application/install-exe-or-msi.adoc [hunk 14]
```diff
@@ -0,0 +1,78 @@
+= Install a software from a .exe or .msi installer.
```
Comment (amousset) [inline rudder-by-example/src/application/install-exe-or-msi.adoc]: remove "a" and the dot at the end

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**